src/main/java/edu/cornell/kfs/pmw/batch/service/impl/PaymentWorksVendorToKfsVendorDetailConversionServiceImpl.java

```diff
@@ -63,7 +65,12 @@ protected KfsVendorDataWrapper createVendorDetail(PaymentWorksVendor pmwVendor,
                                                       Map<String, List<PaymentWorksIsoFipsCountryItem>> paymentWorksIsoToFipsCountryMap, 
                                                       Map<String, SupplierDiversity> paymentWorksToKfsDiversityMap) {
         
-        KfsVendorDataWrapper kfsVendorDataWrapper = populateTaxRuleDependentAttributes(pmwVendor, paymentWorksIsoToFipsCountryMap);
+        KfsVendorDataWrapper kfsVendorDataWrapper = null;
+        if (paymentWorksFormModeService.shouldUseLegacyFormProcessingMode()) {
+            kfsVendorDataWrapper = populateTaxRuleDependentAttributes(pmwVendor, paymentWorksIsoToFipsCountryMap);
+        } else if (paymentWorksFormModeService.shouldUseForeignFormProcessingMode()) {
+            kfsVendorDataWrapper = paymentWorksTaxRuleDependencyService.populateTaxRuleDependentAttributes(pmwVendor, paymentWorksIsoToFipsCountryMap);
+        }
         if (ObjectUtils.isNotNull(kfsVendorDataWrapper.getVendorDetail())) {
```

Comment:
since kfsVendorDataWrapper could be null should there be a null check on kfsVendorDataWrapper first?

Reply:
I altered the code flow, NULL shouldn't happen now.